plugins/backend/catalog-backend-module-okta/src/providers/OktaGroupEntityProvider.ts

```diff
@@ -136,14 +143,28 @@ export class OktaGroupEntityProvider extends OktaEntityProvider {
           group,
           oktaGroups,
         });
-
+        const profileAnnotations: Record<string, string> = {};
+        if (this.customAttributesToAnnotationAllowlist.length) {
+          for (const [key, value] of new Map(Object.entries(group.profile))) {
+            const stringKey = key.toString();
```

Comment:
I'd rather not have its type in the variable name. It should be obvious it is a string after the toString().

We could use let and re-use the key variable.

Reply:
I actually dont think that this line is necessary. The key should always be a string. https://github.com/okta/okta-sdk-nodejs/blob/okta-sdk-nodejs-6.6.0/src/types/models/GroupProfile.d.ts#L27 same with UserProfile.

Reply:
mmm i was pretty sure i did this for a reason but removing it an rerunning the tests just works so i guess i got confused at the moment i wrote this, i'll remove this line